[DOC-992] TA-27177 : PostgreSQL logical replication and v2025.1 upgrade (#28790)
yugabyte/yugabyte-db: docs/content/preview/releases/techadvisories/ta-27177.md

---
title: TA-27177
headerTitle: PostgreSQL logical replication and v2025.1 upgrade
headcontent: 03 October 2025
type: docs
showRightNav: true
cascade:
  unversioned: true
menu:
  preview_releases:
    identifier: ta-27177
    weight: 1
rightNav:
  hideH2: true
type: docs
---

| Product |  Affected Versions  |  Related Issues   | Fixed In |
| :------ | :------------------ | :-------------    | :------- |

> | {{<product "cdc">}} | {{<release "2025.1.0">}} | {{<issue 27177>}} |  |

Reviewer: it is fixed in 2025.1.1

## Description

When you upgrade a YugabyteDB universe from a version running PostgreSQL 11 (all versions prior to v2.25) to a version based on PostgreSQL 15 (v2025.1 or later (stable) or v2.25 or later (preview)), the PostgreSQL catalog tables are deleted and recreated as part of the upgrade process.

> If a logical replication stream was active on the universe before the upgrade and was lagging at the time of the upgrade, such a stream will begin erroring out after the upgrade. This is because logical replication (the walsender process) relies on querying the PostgreSQL catalog tables as of a past time to accurately determine the polling table list and their schemas when the stream is lagging. After the upgrade, the walsender will encounter an error if it attempts to query data from catalog tables pertaining to a time before the upgrade, rendering the replication slot useless.

Reviewer: ```suggestion
If a logical replication stream was active on the universe before the upgrade and was lagging at the time of the upgrade, the stream will begin erroring out after the upgrade. This is because logical replication (the walsender process) relies on querying the PostgreSQL catalog tables as of a past time to accurately determine the polling table list and their schemas when the stream is lagging. After the upgrade, the walsender will encounter an error if it attempts to query data from catalog tables pertaining to a time before the upgrade, rendering the replication slot useless.
```

> This issue only affects the logical replication model of Change Data Capture (CDC). If you use the gRPC model of CDC, you can upgrade to PostgreSQL 15 YugabyteDB versions without any issues.

Reviewer: ```suggestion
This issue only affects Change Data Capture (CDC) using logical replication. If you use CDC with gRPC replication, you can upgrade to PostgreSQL 15 versions of YugabyteDB without any issues.
```
